test/cases/runner-cli/runner.cli.spec.js

```diff
@@ -26,46 +26,52 @@ describe('CLI Fixture', function() {
     destination: path.join(outputPath, 'webcomponents-bundle.js')
   }];
 
-  describe('default options with relative path', function() {
-    let runner;
-
-    before(function() {
-      runner = new Runner();
-      runner.setup(outputPath);
-      runner.runCommand(
-        `${fixturesPath}/cli.js`, // binPath
-        fixturesPath // args
-      );
-    });
-
-    it('should have created the output folder', function() {
-      const exists = fs.existsSync(outputPath);
-
-      expect(exists).to.be.equal(true);
-    });
-
-    it('should only copy 3 files', function() {
-      const files = fs.readdirSync(outputPath);
-
-      expect(files.length).to.be.equal(3);
-    });
-
-    it('should have an .editorconfig file', function() {
-      expect(fs.existsSync(`${outputPath}/.editorconfig`)).to.be.equal(true);
-    });
-
-    it('should have an .eslintrc file', function() {
-      expect(fs.existsSync(`${outputPath}/.eslintrc.cjs`)).to.be.equal(true);
-    });
-
-    it('should have .mocharc file', function() {
-      expect(fs.existsSync(`${outputPath}/.mocharc.cjs`)).to.be.equal(true);
-    });
-
-    it('should delete the output directory when told', function() {
-      runner.teardown([outputPath]);
-
-      expect(fs.existsSync(outputPath)).to.be.equal(false);
+  [
+    {options: {}, label: 'default options with relative path'},
+    {options: { async: true }, label: 'default options with relative path (async)'}
+  ].forEach(({options, label}) => {
+    describe(label, function() {
+      let runner;
+
+      before(async function() {
+        runner = new Runner();
+        await runner.setup(outputPath);
+        await runner.runCommand(
```

Comment:
This is needed to catch any errors that would be thrown by calling `reject` in the Promise executor function. Alternatively, we could stick with the promise chaining paradigm and add `.catch`, but then we'd need to wrap this whole thing in a promise or use the `done()` param in `before`. (https://mochajs.org/#asynchronous-code)

The question is, should we make this change in the `before` method for all our tests?

Reply:
I'm referring to adding `await` before `runner.runCommand`, as well as converting this to an `async` function. I'm just talking about the test cases (`before`). I think all of our other test files (`before`) should be changed to also catch the exception as it now does in this file. For consistency it should match whatever pattern we use here.

My recommendation is we use the async/await pattern, but there are alternatives that I detailed in my last comment. I just want to make sure we agree on which pattern to use before I update all the other test files.